knowledge_agent/prompt.py

```
@@ -0,0 +1,151 @@
+
+        elif intent == IntentCategory.ARCHITECTURE_UNDERSTANDING:
+            if evidence.get("architecture_files"):
+                prompt += "Architecture-related files found:\n" + "\n".join([f"- {p}" for p in evidence["architecture_files"]]) + "\n\n"
+            if evidence.get("tree_sample"):
+                prompt += "Repository structure:\n" + "\n".join([f"- {p}" for p in evidence["tree_sample"]]) + "\n\n"
+
+        elif intent == IntentCategory.REPO_ONBOARDING:
+            if evidence.get("tree"):
+                prompt += f"Repository tree ({len(evidence['tree'])} files):\n" + "\n".join([f"- {p}" for p in evidence["tree"][:25]]) + "\n\n"
+
+        elif "issue" in evidence and evidence["issue"]:
```

Ruff (0.16.1)

[warning] Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)